degica/barcelona
# Add support for filtering Nginx logs

Because there are GDPR requirements on sending Personally Identifiable Information (PII) to third party services, we want to make sure we are not logging any PII that gets sent as query params in requests. This PR uses the [Nginx JS Module](https://www.nginx.com/blog/introduction-nginscript/) to check query params in the incoming requests, and filter out any that match the Email regex.

I see this as a first step in supporting log filtering, if needed I think we can expand it and formalise an API around the JS modules, to make it easier to add new filters or instance specific filters. But I would rather wait until we have a clearer understanding of what we would need before trying to solve that problem 😄 . 

## How to Test
**Testing with filtering enabled**
1. Clone the branch
1. Run `docker-compose -f docker-compose.test.yml build`
1. Run `docker-compose -f docker-compose.test.yml up`
1. Using the browser (or curl) go to http://localhost:50132/?email=john.smith@smith.com
1. In the nginx logs you should see `email=[EMAIL FILTERED]`, like the screenshots

**Testing with filtering disabled**
1. Stop the docker containers session if it's still running
1. Change the [`FILTER_LOGS` param](https://github.com/degica/barcelona/pull/601/files#diff-71027439e4e13f5ebb931f5ea1f97b66R14) to "false" (or remove it entirely)
1. Run `docker-compose -f docker-compose.test.yml up`
1. Using the browser (or curl) go to http://localhost:50132/?email=john.smith@smith.com
1. In the nginx logs you should see `email=john.smith@smith.com`

## Screenshots
**with `FILTER_LOGS=true`**
<img width="1181" alt="Screen Shot 2020-07-20 at 3 08 52 pm" src="https://user-images.githubusercontent.com/12392541/87903520-8187e880-ca9f-11ea-8696-3c9e1efbef46.png">

**with `FILTER_LOGS=false`**
<img width="1189" alt="Screen Shot 2020-07-20 at 3 09 05 pm" src="https://user-images.githubusercontent.com/12392541/87903515-7f258e80-ca9f-11ea-8990-6b108b3a5635.png">
